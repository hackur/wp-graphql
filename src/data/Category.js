import { toGlobalId } from 'graphql-relay';

const path = process.env.WP_CATEGORIES_ENDPOINT || 'wp/v2/categories';

class Category {
  getID() {
    return toGlobalId(this.constructor.name, this.id);
  }

  static getEndpoint() {
    return path;
  }
}

export default Category;
