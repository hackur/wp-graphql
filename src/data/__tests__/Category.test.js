import Category from 'data/Category';

describe('Test Category data access', () => {
  test('Get endpoint', () => {
    expect(Category.getEndpoint()).toMatchSnapshot();
  });
});
