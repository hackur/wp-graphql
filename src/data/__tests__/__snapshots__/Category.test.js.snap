// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Category data access Get endpoint 1`] = `"wp/v2/categories"`;
